### DISCUSSION TITLE
Release of version 12 vs edge

### DISCUSSION THREAD

**Discussion by alexanderadam** [docker-mailserver/docker-mailserver#3122] [Q&A]
I ran into [some bugs that appear in the currently stable version `11.3.1` but not on `edge`](https://github.com/docker-mailserver/docker-mailserver/issues/3110#issuecomment-1445313587).
Surely I don't want to have an unstable or dangerous production environment but I am looking for a solution that I can deploy fast.

Should I go for `edge` for now and switch to `12` once it out?
Or do you have any other advice?

**Comment by casperklein** [comment]
Edge should be fairly stable. In case you encounter any major issue, it will probably fixed really fast.

**Reply by alexanderadam** [reply]
But `edge` will also be part of unstable changes, right?
So it could be that a server using today's `edge` release might behave differently than tomorrow's release, right?

And is there an ETA when the current edge might be released in a stable version release?

**Comment by casperklein** [comment]
> But edge will also be part of unstable changes, right?

It depends on how you define unstable changes. Anything that is merged, is reviewed and tested beforehand. We do not merge things we know that are unreliable (unstable). Thinks will be fixed, when someone running edge encounters an issue. Otherwise, issues may firstly reported after releasing a new version. I don't know, how many people run the edge images, besides the maintainers. What I want to say is, it's not that unlikely, that we miss a bug and release a new version.

> So it could be that a server using today's edge release might behave differently than tomorrow's release, right?

Absolutely.

> And is there an ETA when the current edge might be released in a stable version release?

It's done when it's done 😉 But chances are good, that we release a new version in march/april.

**Reply by alexanderadam** [reply]
Thank you again for your fast response. :raised_hands: 

> Anything that is merged, is reviewed and tested beforehand. We do not merge things we know that are unreliable (unstable).

Of course. Nobody is claiming that anyone would merge something unreviewed but yet here I am asking this question only [because I ran into permission issues with the stable version `11.3.1`](https://github.com/docker-mailserver/docker-mailserver/issues/3110#issuecomment-1445313587) and even tiny mistakes, like [the wrong `/VERSION` label in the container image](https://github.com/docker-mailserver/docker-mailserver/issues/2952) are getting merged unnoticed.

These things just happen of course and it wouldn't be a major issue if I would have a working quick fix so that I have a solution that could work a while stable. Furthermore there's also not a `11.3.2` with backported patches and the 'buggy' release is two months old.

Hence I'm just curious what the recommended options are. :laughing: 

**PS:** Thank you for working on DMS :raised_hands:

**Reply by casperklein** [reply]
> Furthermore there's also not a 11.3.2 with backported patches and the 'buggy' release is two months old.

I agree, a lot of things went wrong.. and I am not very happy with that too.

> Hence I'm just curious what the recommended options are. 😆

If you are brave, go with :edge 🚀 
Otherwise stay on :latest and wait for a new release 🎉

**Reply by polarathene** [reply]
> yet here I am asking this question only [because I ran into permission issues with the stable version `11.3.1`](https://github.com/docker-mailserver/docker-mailserver/issues/3110#issuecomment-1445313587)

The issue with the socket has been reported by a few users, but none stuck around to assist identifying the cause as it only appears to affect a few and unclear how to reproduce. AFAIK the problem would have been present for quite some time, not just our 11.x series of releases.

> and even tiny mistakes, like https://github.com/docker-mailserver/docker-mailserver/issues/2952 are getting merged unnoticed.

That was more to do with how releases are handled (a tag to a git commit) and the `VERSION` file not being updated prior to that tag. We aren't as well setup to push hot fixes unfortunately once breaking changes are merged.

Going forward we could probably better adopt release branches, but that requires time + effort to adjust the project for and is rarely needed. We have had an internal maintainer discussion a while back on how to approach it, but it's not been a priority for any maintainer to tackle with their volunteer time :sweat_smile: 

> Furthermore there's also not a `11.3.2` with backported patches and the 'buggy' release is two months old.

Covered already, backports aren't something we're setup for at present. It'd be great if we could get another maintainer onboard that could assist sorting that out though :)

> These things just happen of course and it wouldn't be a major issue if I would have a working quick fix so that I have a solution that could work a while stable.

If you're new to our project, we usually can get out a new release much sooner. The next release has had considerable refactoring effort going on and many breaking changes being introduced. That should be calming down and I would expect a release should appear in March.

---

> **PS:** Thank you for working on DMS

:heart:
